src/genie/libs/parser/linux/route.py

```diff
@@ -160,3 +153,288 @@ def cli(self, output=None):
 
         return super().cli(output=out)
 
+# =====================================================
+# Parser for ip route show table all
+# =====================================================
+class IpRouteShowTableAll(RouteSchema):
+    """
+    Parser for
+        * ip route show table all
+    """
+
+    cli_command = ['ip route show table all']
+
+    def cli(self, output=None):
+        if output is None:    
+            cmd = self.cli_command[0]
+            out = self.device.execute(cmd)
+        else:
+            out = output
+
+        # default via 192.168.1.1 dev enp7s0 proto dhcp metric 100 
+
+        p1 = re.compile(r'default via (?P<gateway>[a-z0-9\.\:]+)'
```

Comment:
General comment for regular expressions, we usually accommodate for extra spaces by adding **+** after every white space.

Also using **\S** can replace **[a-z0-9\.\:]** which makes it easier to read and maintain.